### PR TITLE
Add option to disable ReduceL2 decomposition

### PR DESCRIPTION
**Pull request by tr00**
Adds an option to disable ReduceL2 decomposition.

### CODE REVIEW

File: src/Dialect/ONNX/Transforms/Decompose.cpp [hunk 1]
```diff
@@ -4253,6 +4253,24 @@ struct DecomposeGroupNormV18Pattern
...
+      ONNXReduceL2Op op, PatternRewriter &rewriter) const final {
+    auto sumSquare = rewriter.create<ONNXReduceSumSquareOp>(op.getLoc(),
+        op.getType(), op.getData(), op.getAxes(), op.getKeepdimsAttr(),
+        op.getNoopWithEmptyAxesAttr());
+    auto sqrtVal = rewriter.create<ONNXSqrtOp>(
+        op.getLoc(), op.getType(), sumSquare.getResult());
+    rewriter.replaceOp(op, sqrtVal.getResult());
+    return success();
+  }
+};
```
Comment (SamNour) [inline src/Dialect/ONNX/Transforms/Decompose.cpp:4272]: Nit:
variables that aren’t modified should be `const`.

File: src/Dialect/ONNX/Transforms/Decompose.cpp [hunk 2]
```diff
@@ -4716,6 +4737,11 @@ struct DecomposeONNXToONNXPass
                      "seq_len>1 LSTM into a chain of seq_len=1 LSTMs)"),
       ::llvm::cl::init(false)};
 
+  Option<bool> enableReduceL2Decompose{*this, "enable-reducel2-decompose",
+      llvm::cl::desc("Enable decomposition of ReduceL2 to "
+                     "Sqrt(ReduceSumSquare(x))"),
+      ::llvm::cl::init(true)};
```
Comment (SamNour) [inline src/Dialect/ONNX/Transforms/Decompose.cpp:4743]: Are we missing tests for this flag? even if the flag is enabled by default we should have negative test cases.

**Review by p-lanza — APPROVED** [review]

File: src/Dialect/ONNX/Transforms/Decompose.cpp (hunk 1 above)
Comment (tr00) [reply src/Dialect/ONNX/Transforms/Decompose.cpp:4272]: getResult() is not available for const onnx ops.

File: src/Dialect/ONNX/Transforms/Decompose.cpp (hunk 2 above)
Comment (tr00) [reply src/Dialect/ONNX/Transforms/Decompose.cpp:4743]: Fair point. I've added some now.

**Review by SamNour — APPROVED** [review]
Thanks!
